File: 5 Longest Palindromic Substring.c
char* longestPalindrome(char* c) {
    int i,j,l,max=0,s,e;
    l=strlen(c);
    for(i=0;i<l;i++){
        for(j=1;j<=1000;j++){
            if(i-j+1<0||i+j-1>=l||c[i-j+1]!=c[i+j-1])break;
            if(j*2-1>max){
                max=j*2-1;
                s=i-j+1;
                e=i+j-1;
            }
        }
        for(j=1;j<=1000;j++){
            if(i-j+1<0||i+j>=l||c[i-j+1]!=c[i+j])break;
            if(j*2>max){
                max=j*2;
                s=i-j+1;
                e=i+j;
            }
        }
    }
    c[e+1]=0;
    return c+s;
}

File: 5 Longest Palindromic Substring.java
public class Solution {
    public static String longestPalindrome(String s) {
        char[] c = s.toCharArray();
        int bs = 0, bv = 1;
        for (int i = 0; i < c.length; i++) {
            int l = i, r = i;
            while (l > 0 && r < c.length - 1 && c[l-1] == c[r+1]) {
                l--;
                r++;
            }
            if (r - l + 1 > bv) {
                bv = r - l + 1;
                bs = l;
            }
            l = i;
            r = i - 1;
            while (l > 0 && r < c.length - 1 && c[l-1] == c[r+1]) {
                l--;
                r++;
            }
            if (r - l + 1 > bv) {
                bv = r - l + 1;
                bs = l;
            }
        }
        return s.substring(bs, bs + bv);
    }
}
